File: spring-security-in-action/ssia-ch07/src/test/java/org/example/ssiach07/controller/HelloControllerTest.java
package org.example.ssiach07.controller;

import static org.junit.jupiter.api.Assertions.*;
import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.*;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
import org.springframework.test.web.servlet.result.MockMvcResultMatchers;

@SpringBootTest
@AutoConfigureMockMvc
class HelloControllerTest {

    @Autowired
    private MockMvc mvc;

    @Test
    void hello() throws Exception {
        mvc.perform(get("/hello").with(httpBasic("writer", "12345")))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$").value("Hello!"));
    }

    @Test
    void 실패테스트() throws Exception {
        mvc.perform(get("/hello").with(httpBasic("reader", "12345")))
                .andExpect(status().isForbidden());
    }
}

File: spring-security-in-action/ssia-ch3/src/test/kotlin/org/example/ssiach3/controller/HelloControllerTest.kt
package org.example.ssiach3.controller

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors
import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.*
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

@SpringBootTest
@AutoConfigureMockMvc
class HelloControllerTest {

    @Autowired
    lateinit var mockMvc: MockMvc

    @Test
    fun hello() {
        mockMvc.perform(
            get("/hello")
                .with(csrf())
                .with(httpBasic("john", "12345"))
        )
            .andExpect(status().isOk)
            .andExpect(jsonPath("$").value("Hello World"))
    }

    @Test
    fun fail() {
        mockMvc.perform(
            get("/hello")
                .with(csrf())
        ).andExpect(status().isUnauthorized())
    }
}